File: barber-app/mobile-flutter/lib/models/barber.dart
class Barber {
  final String id;
  final String name;
  final String? specialty;
  final String? phone;
  final String? email;
  final bool isActive;
  final String barbershopId;

  Barber({
    required this.id,
    required this.name,
    this.specialty,
    this.phone,
    this.email,
    required this.isActive,
    required this.barbershopId,
  });

  bool get active => isActive;

  factory Barber.fromJson(Map<String, dynamic> json) {
    return Barber(
      id: json['id'].toString(),
      name: json['name'] ?? '',
      specialty: json['specialty'],
      phone: json['phone'],
      email: json['email'],
      isActive: json['isActive'] ?? true,
      barbershopId: json['barbershopId']?.toString() ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'specialty': specialty,
      'phone': phone,
      'email': email,
      'isActive': isActive,
      'barbershopId': barbershopId,
    };
  }
}

File: barber-app/barber-app/src/main/java/com/barber/domain/entity/Barber.java
package com.barber.domain.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "barbers")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Barber {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    
    @Column(nullable = false)
    private String name;
    
    private String specialty;
    
    private String phone;
    
    private String email;
    
    @Column(name = "is_active", nullable = false)
    private Boolean isActive;
    
    @Column(name = "barbershop_id", nullable = false)
    private String barbershopId;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "barbershop_id", insertable = false, updatable = false)
    private Barbershop barbershop;
    
    @PrePersist
    protected void onCreate() {
        if (isActive == null) {
            isActive = true;
        }
    }
}
